spec/datadog/tracing/integration_spec.rb

```diff
@@ -577,6 +599,81 @@ def sample!(trace)
         expect(stats[:transport].internal_error).to eq(0)
       end
     end
+
+    context 'with agent rates' do
+      before do
+        WebMock.enable!
+        stub_request(:post, %r{/v0.4/traces}).to_return(status: 200, body: service_rates.to_json)
+      end
+
+      after { WebMock.disable! }
+
+      let(:service_rates) { { rate_by_service: { 'service:kept,env:' => 1.0, 'service:dropped,env:' => Float::MIN } } }
+
+      let(:set_agent_rates!) do
+        # Send span to receive response from "agent" with mocked service rates above.
+        tracer.trace('send_trace_to_fetch_service_rates') {}
+        try_wait_until { tracer.writer.stats[:traces_flushed] >= 1 }
+
+        # Reset stats and collected segments before test starts
+        tracer.writer.send(:reset_stats!)
+        trace_segments.clear
+      end
+
+      context 'without DD_ENV set' do
+        before { set_agent_rates! }
+
+        context 'with a kept trace' do
+          before do
+            tracer.trace('kept.span', service: 'kept') {}
+            try_wait_until { tracer.writer.stats[:traces_flushed] >= 1 }
+          end
+
+          it_behaves_like 'priority sampled', 1.0
+        end
+
+        context 'with a dropped span' do
+          before do
+            tracer.trace('dropped.span', service: 'dropped') {}
+            try_wait_until { tracer.writer.stats[:traces_flushed] >= 1 }
+          end
+
+          it_behaves_like 'priority sampled', 0.0
+        end
+      end
+
+      context 'with DD_ENV set' do
+        before do
+          Datadog.configure do |c|
+            c.env = 'test'
+          end
+
+          set_agent_rates!
+        end
+
+        let(:service_rates) do
+          { rate_by_service: { 'service:kept,env:test' => 1.0, 'service:dropped,env:' => Float::MIN } }
+        end
+
+        context 'with a span matching the environment rates' do
+          before do
+            tracer.trace('kept.span', service: 'kept') {}
+            try_wait_until { tracer.writer.stats[:traces_flushed] >= 1 }
+          end
+
+          it_behaves_like 'priority sampled', 1.0
+        end
+
+        context 'with a span not matching the environment rates' do
+          before do
+            tracer.trace('kept.span', service: 'kept', tags: { 'env' => 'not-right' }) {}
+            try_wait_until { tracer.writer.stats[:traces_flushed] >= 1 }
+          end
+
+          it_behaves_like 'priority sampled', 1.0
+        end
```

Comment:
I... don't think we're testing the right thing here? In particular, I reverted the change to `rule_sampler.rb` and the test still passed?